sormas-backend/src/main/java/de/symeda/sormas/backend/selfreport/SelfReportImportFacadeEjb.java

```diff
@@ -0,0 +1,311 @@
+
+import javax.ejb.EJB;
+import javax.ejb.LocalBean;
+import javax.ejb.Stateless;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.symeda.sormas.api.Language;
+import de.symeda.sormas.api.i18n.Captions;
+import de.symeda.sormas.api.i18n.I18nProperties;
+import de.symeda.sormas.api.i18n.Validations;
+import de.symeda.sormas.api.importexport.ImportCellData;
+import de.symeda.sormas.api.importexport.ImportErrorException;
+import de.symeda.sormas.api.importexport.ImportLineResultDto;
+import de.symeda.sormas.api.importexport.InvalidColumnException;
+import de.symeda.sormas.api.infrastructure.community.CommunityReferenceDto;
+import de.symeda.sormas.api.infrastructure.district.DistrictReferenceDto;
+import de.symeda.sormas.api.infrastructure.facility.FacilityReferenceDto;
+import de.symeda.sormas.api.selfreport.SelfReportDto;
+import de.symeda.sormas.api.selfreport.SelfReportImportFacade;
+import de.symeda.sormas.api.selfreport.SelfReportType;
+import de.symeda.sormas.api.user.UserRight;
+import de.symeda.sormas.api.utils.DataHelper;
+import de.symeda.sormas.api.utils.DateHelper;
+import de.symeda.sormas.api.utils.ValidationRuntimeException;
+import de.symeda.sormas.backend.importexport.ImportFacadeEjb;
+import de.symeda.sormas.backend.importexport.ImportHelper;
+import de.symeda.sormas.backend.infrastructure.community.CommunityFacadeEjb;
+import de.symeda.sormas.backend.infrastructure.district.DistrictFacadeEjb;
+import de.symeda.sormas.backend.infrastructure.facility.FacilityFacadeEjb;
+import de.symeda.sormas.backend.util.RightsAllowed;
+
+@Stateless(name = "SelfReportImportFacade")
+@RightsAllowed(UserRight._SELF_REPORT_IMPORT)
+public class SelfReportImportFacadeEjb implements SelfReportImportFacade {
+
+	private static final String ERROR_COLUMN_NAME = I18nProperties.getCaption(Captions.importErrorDescription);
+	private final Logger LOGGER = LoggerFactory.getLogger(getClass());
+
+	@EJB
+	private SelfReportFacadeEjb.SelfReportFacadeEjbLocal selfReportFacade;
+	@EJB
+	private ImportFacadeEjb.ImportFacadeEjbLocal importFacade;
+	@EJB
+	private DistrictFacadeEjb.DistrictFacadeEjbLocal districtFacade;
+	@EJB
+	private CommunityFacadeEjb.CommunityFacadeEjbLocal communityFacade;
+	@EJB
+	private FacilityFacadeEjb.FacilityFacadeEjbLocal facilityFacade;
+
+	@Override
+	public ImportLineResultDto<SelfReportDto> importSelfReportData(
+		String[] values,
+		String[] entityClasses,
+		String[] entityProperties,
+		String[][] entityPropertyPaths,
+		boolean ignoreEmptyEntries) {
+
+		// Check whether the new line has the same length as the header line
+		if (values.length > entityProperties.length) {
+			return ImportLineResultDto.errorResult(I18nProperties.getValidationError(Validations.importLineTooLong));
+		}
+
+		SelfReportDto selfReport;
+		if (values[0].isBlank() || !EnumUtils.isValidEnum(SelfReportType.class, values[0])) {
```

Comment:
having the type in the first column is not a must, there is no need to check it just do a build with null all the time